test/fetch/content-length.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+const { test } = require('tap')
+const { Blob } = require('buffer')
+const { FormData, fetch } = require('../..')
+const { once } = require('events')
+const { createServer } = require('http')
+
+const isV16x = process.version.startsWith('v16.')
+
+// https://github.com/nodejs/undici/issues/1783
+test('Sending a FormData body sets Content-Length header', { skip: isV16x }, async (t) => {
```

Comment:
why skipped on node 16?

Reply:
The length is different on ubuntu v16.x for some reason (3 extra characters somewhere).

Reply:
> The length is different on ubuntu v16.x for some reason (3 extra characters somewhere).

Could it be the random identifier in boundary?

Reply:
yes, most likely. Although locally tests passed in v16 so I decided just to disable this one as there are other tests that run and will fail if there's an issue with the content-length calculation in the future.

Reply:
I would prefer if we understood what's going on here

Reply:
A weird ubuntu-only ci-only (?) 'bug' where the length calculated is different than other platforms. The length was calculated correctly and the body was sent, otherwise we'd get an error for mismatched content-length.

Reply:
How can the length be calculated differently?

Reply:
maybe you are calling `Math.random()` twice? once when you are calculating the length and another time when you are sending the bits to the api endpoint?

Anyway Math.random() can return different length.
```js
i=1000; while(i--) console.log(Math.random().toString().length) // 17-20 can vary more...
```

> I considered doing this but decided against it because we'd have to load the entirety of the FormData into memory first.

A good Blob impl wouldn't have to read everything into memory if it can just make references points to other blob parts.
so: `new Blob([file1, file2])` would only hold 2 references points to other blob barts without reading anything into memory basically.
only the strings would allocate new memory. (btw, doing it as you have currently done requires to basically do the same logic twice `escape + normalizeLinefeeds` where as a formData to blob would already have pre computed the data that needs to be sent. 

The history of Deno's formData to blob originates from my formdata package where the old browsers wouldn't be able to write a stream to the request (so had to have all data ready at `xhr.send(blob)`

I think it's a decent solution...

Reply:
In my formdata impl (where i did generate the boundary i did something like 
```js
`${ Math.random() }${ Math.random() }`.slice(-28).padStart(32, '-')`
```
so i always had the same length...

Reply:
I'll remove this test since we have coverage for sending POST requests with FormData bodies already, since it relies on a length which can/might vary(?).

Reply:
Why not fix it?

Reply:
Nothing is really broken - otherwise we'd get a content-length mismatch error. We also have both normal tests and WPTs that ensure that the length is being calculated properly.

Maybe in a future PR we could use a fixed length boundary.